File: Docker/programs/benchmark/python/LCG.py
import math

class LCG:
    def __init__(self, seed):
        self.state = self.__hash_seed(seed)
        self.a = 1664525
        self.c = 1013904223
        self.m = 1294967296

    def __next(self):
        self.state = (self.a * self.state + self.c) % self.m
        return self.state

    def random(self):
        return self.__next() / float(self.m)

    def get_next_number(self,n):
        return math.floor(self.random() * n)

    def get_next_number_between(self,n,m):
        return math.floor(self.random() * (m-n))+n

    def __hash_seed(self,input_str):
        prime = 31
        checksum = 0

        for char in input_str:
            checksum = (checksum * prime + ord(char)) & 0xFFFFFFFF

        return checksum

File: Docker/programs/benchmark/java/src/main/java/org/betriebssysteme/LCG.java
package org.betriebssysteme;

class LCG {
    private long state;
    private final int a = 1664525;
    private final int c = 1013904223;
    private final long m = 1294967296;

    public LCG(String seed) {
        this.state = hashSeed(seed);
    }

    private long hashSeed(String inputStr) {
        int prime = 31;
        long checksum = 0;

        for (char ch : inputStr.toCharArray()) {
            checksum = (checksum * prime + (int) ch) & 0xFFFFFFFFL;
        }
        return checksum;
    }

    private long next() {
        state = (a * state + c) % m;
        return state;
    }

    public double random() {
        return (double) next() / m;
    }

    public int getNextNumber(int n) {
        return  (int)Math.round(Math.floor(random() * n));
    }

    public int getNextNumberBetween(int n, int m) {
        return  (int)Math.round(Math.floor(random() * (m - n)) + n);
    }
}